support-frontend/app/controllers/RedemptionController.scala

```diff
@@ -50,26 +57,46 @@ class RedemptionController(
   val js = "subscriptionsRedemptionPage.js"
   val css = "digitalSubscriptionCheckoutPage.css" //TODO: Don't need this?
 
-  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = googleAuthAction.async {
+  val getCorporateCustomer = new GetCorporateCustomer(dynamoLookup)
+  val testUserFromRequest = new TestUserFromRequest(testUsers)
+
+  def displayForm(redemptionCode: RawRedemptionCode): Action[AnyContent] = (googleAuthAction andThen maybeAuthenticatedAction()).async {
     implicit request =>
-      getCorporateCustomer(redemptionCode).fold(Some.apply, _ => None).map(
-        maybeError =>
-          Ok(subscriptionRedemptionForm(
-            title = title,
-            mainElement = id,
-            js = js,
-            css = css,
-            fontLoaderBundle = fontLoaderBundle,
-            csrf = None,
-            uatMode = false,
-            stage = "checkout",
-            redemptionCode = redemptionCode,
-            maybeRedemptionError = maybeError,
-            user = None,
-            submitted = false
-          )).withHeaders(CacheControl.noCache)
-      )
-    }
+      for {
+        whyIsTestUser <- request.user.map { authenticatedIdUser =>
+          identityService.getUser(authenticatedIdUser.minimalUser)
+        } match {
+          case None => Future.successful(testUserFromRequest.fromCookies(request.cookies))
+          case Some(eventualErrorOrUser) =>
+            eventualErrorOrUser.fold(
+              message => {
+                SafeLogger.error(scrub"could not fetch user - assuming normal backend: $message")
+                None
+              },
+              testUserFromRequest.fromIdUser
+            )
+        }
+        normalisedCode = redemptionCode.toUpperCase(Locale.UK)
```

Comment:
could take this out, I just thought it's nice that the code is displayed in caps on first load

Reply:
Yes worth leaving in I think